File: trail2/숫자 차례로 출력하기/to-output-numerically.py
n = int(input())

# Please write your code here.

def func1(num):
    if num == 0:
        return
    
    func1(num - 1)
    print(num, end=" ")

def func2(num):
    if num == 0:
        return
    
    print(num, end=" ")
    func2(num - 1)

func1(n)
print()
func2(n)

File: trail2/숫자 차례로 출력하기/to-output-numerically.java
import java.io.*;

public class Main {
    static StringBuilder sb = new StringBuilder();
    static void f(int n) {
        if (n == 0) return;
        
        f(n-1);
        sb.append(String.valueOf(n)).append(' ');
    }

    static void g(int n) {
        if (n == 0) return;
        sb.append(String.valueOf(n)).append(' ');
        g(n-1);
    }

    public static void main(String[] args) throws IOException {
        // Please write your code here.
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        f(n);
        sb.append('\n');
        g(n);

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
}